bin/tx-prover/src/proxy/worker.rs

```diff
@@ -0,0 +1,102 @@
+// WORKER
+// ================================================================================================
+
+use std::time::Duration;
+
+use pingora::lb::Backend;
+use tonic::transport::Channel;
+use tonic_health::pb::{
+    health_check_response::ServingStatus, health_client::HealthClient, HealthCheckRequest,
+};
+use tracing::error;
+
+use crate::commands::WorkerConfig;
+
+/// A worker used for processing of requests.
+///
+/// A worker consists of a backend service (defined by worker address), a flag indicating wheter
+/// the worker is currently available to process new requests, and a gRPC health check client.
+#[derive(Debug, Clone)]
+pub struct Worker {
+    pub(crate) worker: Backend,
+    health_check_client: HealthClient<Channel>,
+    pub(crate) is_available: bool,
+}
+
+impl Worker {
+    pub async fn new(
+        worker: Backend,
+        connection_timeout: Duration,
+        total_timeout: Duration,
+    ) -> Self {
+        let health_check_client = Self::create_health_check_client(
+            worker.addr.to_string(),
+            connection_timeout,
+            total_timeout,
+        )
+        .await
+        .expect("Could not create health check client");
+
+        Self {
+            worker,
+            is_available: true,
+            health_check_client,
+        }
+    }
+
+    /// Create a gRPC [HealthClient] for the given worker address.
+    ///
+    /// It will panic if the worker URI is invalid.
+    async fn create_health_check_client(
```

Comment:
nit: since this does not use anything from `Self` - I'd probably make it a stand-alone helper function.